### DISCUSSION TITLE
<title>Cloning Repo Error: cygwin DLL

### DISCUSSION THREAD

**Discussion by oohookoo** [community/community#49650] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I keep getting this error when trying to clone a repository. I've tried updating, reinstalling, different repositories, etc. same error. I don't get it. If anyone has any ideas, please let me know. Thanks.

Cloning into 'C:\GitHub\DSAnimStudio'...
remote: Enumerating objects: 6414, done.        
remote: Counting objects: 100% (8/8), done.        
remote: Compressing objects: 100% (8/8), done.        
remote: Total 6414 (delta 0), reused 2 (delta 0), pack-reused 6406        
Receiving objects: 100% (6414/6414), 270.40 MiB | 34.97 MiB/s, done.
Resolving deltas: 100% (3510/3510), done.
Updating files: 100% (2540/2540), done.
      1 [main] sh 446 dofork: child -1 - forked process 27724 died unexpectedly, retry 0, exit code 0xC0000142, errno 11
C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\mingw64\libexec\git-core\git-submodule: fork: retry: Resource temporarily unavailable
1118719 [main] sh 446 dofork: child -1 - forked process 23536 died unexpectedly, retry 0, exit code 0xC0000142, errno 11
C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\mingw64\libexec\git-core\git-submodule: fork: retry: Resource temporarily unavailable
      1 [main] sh (28944) C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\usr\bin\sh.exe: *** fatal error - cygheap base mismatch detected - 0x1571408/0x13F1408.
This problem is probably due to using incompatible versions of the cygwin DLL.
Search for cygwin1.dll using the Windows Start->Find/Search facility
and delete all but the most recent version.  The most recent version *should*
reside in x:\cygwin\bin, where 'x' is the drive on which you have
installed the cygwin distribution.  Rebooting is also suggested if you
are unable to find another cygwin DLL.
3241138 [main] sh 446 dofork: child -1 - forked process 28944 died unexpectedly, retry 0, exit code 0xC0000142, errno 11
C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\mingw64\libexec\git-core\git-submodule: fork: retry: Resource temporarily unavailable
      1 [main] sh (28528) C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\usr\bin\sh.exe: *** fatal error - cygheap base mismatch detected - 0x1571408/0x1491408.
This problem is probably due to using incompatible versions of the cygwin DLL.
Search for cygwin1.dll using the Windows Start->Find/Search facility
and delete all but the most recent version.  The most recent version *should*
reside in x:\cygwin\bin, where 'x' is the drive on which you have
installed the cygwin distribution.  Rebooting is also suggested if you
are unable to find another cygwin DLL.
7359679 [main] sh 446 dofork: child -1 - forked process 28528 died unexpectedly, retry 0, exit code 0xC0000142, errno 11
C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\mingw64\libexec\git-core\git-submodule: fork: retry: Resource temporarily unavailable
      1 [main] sh (22308) C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\usr\bin\sh.exe: *** fatal error - cygheap base mismatch detected - 0x1571408/0x16B1408.
This problem is probably due to using incompatible versions of the cygwin DLL.
Search for cygwin1.dll using the Windows Start->Find/Search facility
and delete all but the most recent version.  The most recent version *should*
reside in x:\cygwin\bin, where 'x' is the drive on which you have
installed the cygwin distribution.  Rebooting is also suggested if you
are unable to find another cygwin DLL.
15471718 [main] sh 446 dofork: child -1 - forked process 22308 died unexpectedly, retry 0, exit code 0xC0000142, errno 11
C:\Users\admin\AppData\Local\GitHubDesktop\app-3.2.0\resources\app\git\mingw64\libexec\git-core\git-submodule: fork: Resource temporarily unavailable

**Comment by Abinashbunty** [comment]
Have you done the steps mentioned towards the end of your question where it says to look for _cygwin1.dll_ on your system and deleting all other versions except the latest one?

**Reply by oohookoo** [reply]
yes and a search only finds one file

Also I find it strange that github desktop would require the use of a program outside of its directory... I'm talking about 'x:\cygwin\bin'

**Reply by Abinashbunty** [reply]
Alright. Then you might have to install git on Cygwin correctly first and then do the cloning within the terminal. 

You can follow this link for the steps. https://www.geeksforgeeks.org/how-to-install-git-on-cygwin/

**Reply by Abinashbunty** [reply]
> Alright. Then you might have to install git on Cygwin correctly first and then do the cloning within the terminal.
> 
> You can follow this link for the steps. https://www.geeksforgeeks.org/how-to-install-git-on-cygwin/

Remove existing installation of git from your system first though.

**Reply by oohookoo** [reply]
okay thank you I will try that. So you have that directory on your system?  cygwin

**Reply by Abinashbunty** [reply]
No, I don't have cygwin on my system. I always use GitHub Desktop and never had come across this. By any chance can you tell which repo are you trying to clone? If it's a public one, I can try on my system as well.
